app/services/suite_selector.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+#
+# Service object to select suites for a given group.
+class SuiteSelector
+  # Allow for :select to be called on the parent class
+  def self.select(**params)
+    new(**params).select
+  end
+
+  # Initialize a new SuiteSelector
+  #
+  # @param group [Group] the group to assign the suite to
+  # @param suite_id [Integer] the suite id to assign to the group
+  def initialize(group:, suite_id:)
+    @group = group
+    @suite_id = suite_id
+    @errors = []
+  end
+
+  # Select / assign a suite to a group. Checks to make sure that the grou does
+  # not curretly have a suite assigned, that the suite_id corresponds to an
+  # existing suite, and that the suite is not currently assigned to a different
+  # group.
+  #
+  # @return [Hash{symbol=>Group,Hash}] a results hash with a message to set in
+  #   the flash, nil or the group as the :object, and an action to render.
+  def select
+    if valid? && assign_suite_to_group
```

Comment:
We use `if user.save` a lot... is there a reason why this would be different? We can't put the `assign_suite_...` method first since that actually calls `group.update` and the validations aren't database- or model-backed.